js/settings.js

```diff
@@ -104,19 +104,18 @@ var Settings = (function () {
         const getUnitDisplayTypeValue = () => {
             // Try and match the values 
             switch (configUnitType) {
-                case UnitType.OSD: // Match the OSD value on the UI
-                    return globalSettings.osdUnits;
+                case UnitType.none:
+                    return  5;
+                    break;
+                case UnitType.metric:
+                    return 0;
                     break;
                 case UnitType.imperial:
-                    return 0; // Imperial OSD Value
+                    return 1;
                     break;
-                case UnitType.metric:
-                    return 1; // Metric + MPH OSD Value
+                case UnitType.OSD: // Match the OSD value on the UI
+                    return globalSettings.osdUnits;
                     break;
-                case UnitType.none:
-                default:
-                    // Something went wrong
-                    return -1;
             }
```

Comment:
Lines 106 -119: This block shouldn't be changed. As the IDs match up to the OSD IDs set by the firmware. Because there is an option to use the OSD selected units, they must sync.

IDs are:
0 - OSD_UNIT_IMPERIAL
1 - OSD_UNIT_METRIC
2 - OSD_UNIT_METRIC_MPH
3 - OSD_UNIT_UK
4 - OSD_UNIT_GA

Additional OSD unit types would use 5 and onwards. This only leaves -1 for no units selected.

Reply:
Imperial and Metric still have the wrong IDs.